agent/config/runtime.go

```diff
@@ -29,6 +29,77 @@ type RuntimeSOAConfig struct {
 	Minttl  uint32 // 0,
 }
 
+// StaticRuntimeConfig specifies the subset of configuration the consul agent actually
+// uses and that are not reloadable by configuration auto reload.
+type StaticRuntimeConfig struct {
```

Comment:
Just a heads up that we're shortly going to be splitting the TLS settings here (e.g. `CertFile`) out by listener (e.g. `GRPC.CertFile`, `HTTPS.CertFile`).

Your PR will probably get merged first, so no immediate action required, but I'll add you as a reviewer on mine to make sure I'm doing it right!

Reply:
@boxofrad do you think the approach I took would make sense with the splitting? Are you planning to split those config in different sub struct in the RuntimeConfig or just split the fields?

Reply:
I need to take a proper look to say for sure, let me get back to you! Plan is for `RuntimeConfig` to have a `tlsutil.Config` struct, with members for each listener `InternalRPC`, `GRPC` and `HTTPS`.

Reply:
I think it should be fine 🤞🏻 (we'll just put the `tlsutil.Config` field on `StaticRuntimeConfig` instead) but there will be more cert files to watch 😄